File: playground/js/src/protocol-kit/deploy-safe.ts
import * as App from "../wrap";
import { SALT_NONCE, getClient } from "../utils";

export const deploySafe = async () => {
  const client = getClient();
  const safe = new App.Safe(client);
  const ethers = new App.Ethers(client);
  // Get signer address
  const signerAddress = await ethers.getSignerAddress({});

  if (!signerAddress.ok) throw signerAddress.error;
  console.log(`Signer address: ${signerAddress.value}`);

  const deploymentInput = {
    safeAccountConfig: {
      owners: [signerAddress.value],
      threshold: 1,
    },
    safeDeploymentConfig: {
      saltNonce: SALT_NONCE,
    },
  };

  const expectedSafeAddress = await safe.predictSafeAddress({
    input: deploymentInput,
  });
  if (!expectedSafeAddress.ok) throw expectedSafeAddress.error;
  console.log(`Expected safe address: ${expectedSafeAddress.value}`);

  console.log("Deploying safe...");

  const deploySafe = await safe.predictSafeAddress({
    input: deploymentInput,
  });
  if (!deploySafe.ok) throw deploySafe.error;

  console.log(`Safe deployed to address: ${deploySafe.value}`);
};

File: playground/rs/src/protocol-kit/deploy-safe.rs
extern crate safe_core_wrap_playground;
use safe_core_wrap_playground::{
    get_client, Ethers, EthersArgsGetSignerAddress, InvokeOptions, Safe, SafeArgsDeploySafe,
    SafeDeploymentInput, SafeEthersConnection, SafeSafeAccountConfig as SafeAccountConfig,
    SafeSafeDeploymentConfig as SafeDeploymentConfig, NETWORK, SALT_NONCE,
};

pub fn main() {
    let client = get_client();

    let invoke_options = InvokeOptions {
        uri: None,
        client: Some(client),
        env: None,
    };

    let ethers = Ethers::new(Some(invoke_options.clone()));
    let safe = Safe::new(Some(invoke_options.clone()));

    let signer_address =
        ethers.get_signer_address(&EthersArgsGetSignerAddress { connection: None }, None);

    println!(
        "Signer address fetched: {}",
        signer_address.clone().unwrap()
    );
    let deployment_input = SafeDeploymentInput {
        safe_account_config: SafeAccountConfig {
            owners: vec![signer_address.unwrap()],
            threshold: 1,
            to: None,
            data: None,
            fallback_handler: None,
            payment_token: None,
            payment: None,
            payment_receiver: None,
        },
        safe_deployment_config: Some(SafeDeploymentConfig {
            salt_nonce: SALT_NONCE.to_string(),
            is_l1_safe: None,
            version: None,
        }),
        connection: Some(SafeEthersConnection {
            network_name_or_chain_id: Some(NETWORK.clone().to_string()),
            node: None,
        }),
        custom_contract_addresses: None,
    };
    let deploy = safe.deploy_safe(
        &SafeArgsDeploySafe {
            input: deployment_input,
            tx_options: None,
        },
        None,
    );

    println!("Safe deployed to the address: {}", deploy.unwrap());
}
